float64[] joint_data